tests/shardddl4_1/run.sh

```diff
@@ -667,17 +668,34 @@ function DM_150_CASE {
 	run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,\"bbbbbbb\");"
 	run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,\"ccccccc\");"
 
-	run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
-		"query-status test" \
-		"Unsupported modify column: length 10 is less than origin 20" 1
+	if [[ "$shardmode" == "pessimistic" ]]; then
+		# ddl: "modify column a varchar(10);" passes in worker1, but in pessimistic mode is still waiting for the other worker in the sharding group to be executed with the same ddl.
+		run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+			"query-status test" \
+			'"ALTER TABLE `shardddl`.`tb` MODIFY COLUMN `a` VARCHAR(10)"' 2
+		run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+			"show-ddl-locks" \
+			'ALTER TABLE `shardddl`.`tb` MODIFY COLUMN `a` VARCHAR(10)"' 1
+
+		# we alter database in source2 and the ddl lock will be resolved
+		run_sql_source2 "alter table ${shardddl1}.${tb1} modify column a varchar(10);"
+		run_sql_source2 "alter table ${shardddl1}.${tb2} modify column a varchar(10);"
+		check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+	else
+		# ddl: "modify column a varchar(10)" is passed in optimistic mode and will be executed downstream.
+		run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
```

Comment:
Optimistic can also use check_sync_diff?

Reply:
Optimistic mode does not pass, the task has been paused